### PR TITLE
[t137957] Improvement coverage trend summarizing changes of score

### PR DESCRIPTION
**Pull request by BT-tkarpinski**
…s updated

 
<div> Links to Odoo: </div> <ul>
<li><a target="_blank" href="https://braintec.com/web#view_type=form&model=project.task&id=137957">[t137957] Improvement coverage trend summarizing changes of score</a></li>
</ul>

### CODE REVIEW

File: .github/workflows/bt_ci_workflow_unit_tests.yml [hunk 1]
```diff
@@ -205,7 +205,7 @@ jobs:
         if: contains(inputs.unit_tests_check_command, 'coverage') && github.event_name == 'push'
         uses: actions/checkout@v3
         with:
-          ref: master
+          ref: BT-tkarpinski-patch-2
```
Comment (BT-tkarpinski) [inline .github/workflows/bt_ci_workflow_unit_tests.yml:208]: Revert to master

File: .github/scripts/update_coverage_badge_readmes.py [hunk 2]
```diff
@@ -83,15 +84,29 @@ def get_coverage_trend(readme_path, coverage_percentage):
...
 
-    labels = str(dates[::-1]).replace(" ", "")
-    data = str(coverage[::-1]).replace(" ", "").replace("'", "")
+    # revert coverage and dates to have them from oldest to newest
+    dates = dates[::-1]
+    coverage = coverage[::-1]
+    # remove duplicates to make the graph more readable and to avoid unnecessary commits
+    ids_to_remove = []
+    prev = None
+    for idc, c in enumerate(coverage):
```
Comment (BT-rmartin) [inline .github/scripts/update_coverage_badge_readmes.py:93]: [minor] Improve readibility with better variable names?

File: .github/scripts/update_coverage_badge_readmes.py [hunk 3]
```diff
@@ -83,15 +84,29 @@ def get_coverage_trend(readme_path, coverage_percentage):
                     coverage.append(0)
                 dates.append(commit_date.strftime('%d-%m-%Y'))
 
-    labels = str(dates[::-1]).replace(" ", "")
-    data = str(coverage[::-1]).replace(" ", "").replace("'", "")
+    # revert coverage and dates to have them from oldest to newest
+    dates = dates[::-1]
+    coverage = coverage[::-1]
+    # remove duplicates to make the graph more readable and to avoid unnecessary commits
```
Comment (BT-rmartin) [inline .github/scripts/update_coverage_badge_readmes.py:90]: [minor] Add "leaving the oldest date in which a coverage percentage was achieved"

**Review by BT-rmartin — APPROVED** [review]
Really good job! Just suggestions of small further improvements, but already approved

**Pull request merged**